### ISSUE TITLE
Doesn't work for non-staged Versioned classes

### ISSUE THREAD

**Issue by jonom** [axllent/silverstripe-version-truncator#16]
Hi @axllent, just trying this module out but found it errors out if you use the Versioned extension just for tracking changes (not for draft/published stages). [See the docs entry about it here](https://docs.silverstripe.org/en/4/developer_guides/model/versioning/#applying-the-versioned-extension-to-your-dataobject)

I found that if I change `TruncateVersionsTask.php::_prune()` to add a `hasStages()` check before `isLiveVersion()` it doesn't crash on these classes, but of course pruning will be skipped.

```php
if ($r->hasStages() && $r->isLiveVersion()) {
    $deleted += $r->doVersionCleanup();
}
```

Adding this quick fix is probably a good idea, but would be cool if this module could prune versions on non-staged classes as well :)

**Comment by axllent** [comment]
Hi @jonom. That's a very good question - in fact I had actually never realised that staging could be disabled when still using Versioned (until now...)!

I've just tested here, though I am not getting any errors - it just doesn't truncate anything. This is being caught (prevented) by [this block](https://github.com/axllent/silverstripe-version-truncator/blob/master/src/VersionTruncator.php#L39-L42).

What version are you running?

**Comment by axllent** [comment]
Oh sorry - I see you are referring to the task, not the auto-truncation.  There is actually a "fix" ([link](https://github.com/axllent/silverstripe-version-truncator/commit/35d3c7d9a4eefe04063468bfb8736bb21e0143a5)) for this which never got released. I say a "fix" because it is effectively the same as what you proposed, so more a "skip".

Would you mind please testing with the master branch and letting me know if that resolves your error `composer require axllent/silverstripe-version-truncator:dev-master` ?

**Comment by axllent** [comment]
This functionality was included recently in 3.0.1. Please feel free to re-open this if you still experience the issue.

**Comment by jonom** [comment]
Thanks @axllent! Sorry, was meaning to get back to you after I tested it but that project has been delayed.
